lib/src/main/java/com/otaliastudios/transcoder/source/DefaultDataSource.java

```diff
@@ -192,6 +194,13 @@ public void releaseTrack(@NonNull TrackType type) {
         }
     }
 
+    @Override
+    public Bitmap getFrameAtPosition(long positionUs) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
+            return mMetadata.getScaledFrameAtTime(positionUs, MediaMetadataRetriever.OPTION_CLOSEST_SYNC, 150, 150);
+        }
+        return mMetadata.getFrameAtTime(positionUs);
```

Comment:
we also need to scale down the image for OS below O

Reply:
Done